pattern/web/__init__.py

```diff
@@ -3675,6 +3675,8 @@ def __init__(self, s):
         s = s.replace(":", " :")        # :pseudo-element
         s = s.replace("[", " [")        # [attribute="value"]
         s = re.sub(r"\[.*?\]",
+                   lambda m: re.sub(r" (\#|\.|\:)", "\\1", m.group(0)), s)
```

Comment:
I think it's this additional line that causes the selector failure (the rest of this commit looks harmless). I do no think I screwed up the merge, I think this is intentionally a new insert. @tom-de-smedt